core/src/main/java/io/aiven/klaw/helpers/db/rdbms/SelectDataJdbc.java

```diff
@@ -1556,16 +1556,39 @@ public int findAllComponentsCountForTeam(Integer teamId, int tenantId) {
     return calculateComponentsCountForTeam(teamId, tenantId, false);
   }
 
+  public boolean existsComponentsCountForUser(String userId, int tenantId) {
+    return calculateComponentsCountForUser(userId, tenantId, true) > 0;
+  }
+
+  public int calculateComponentsCountForUser(String userId, int tenantId, boolean existsOnly) {
+    List<Supplier<Integer>> list =
+        List.of(
+            () ->
+                ((Long) schemaRequestRepo.findAllRecordsCountForUserId(userId, tenantId).get(0)[0])
+                    .intValue(),
+            () ->
+                ((Long)
+                        kafkaConnectorRequestsRepo.findAllRecordsCountForUserId(userId, tenantId)
+                            .get(0)[0])
+                    .intValue(),
+            () ->
+                ((Long) topicRequestsRepo.findAllRecordsCountForUserId(userId, tenantId).get(0)[0])
+                    .intValue(),
+            () ->
+                ((Long) aclRequestsRepo.findAllRecordsCountForUserId(userId, tenantId).get(0)[0])
+                    .intValue());
+    int res = 0;
+    for (var elem : list) {
+      res += elem.get();
+      if (existsOnly && res > 0) {
+        return res;
+      }
+    }
+    return res;
+  }
+
   public int findAllComponentsCountForUser(String userId, int tenantId) {
-    return ((Long) schemaRequestRepo.findAllRecordsCountForUserId(userId, tenantId).get(0)[0])
-            .intValue()
-        + ((Long)
-                kafkaConnectorRequestsRepo.findAllRecordsCountForUserId(userId, tenantId).get(0)[0])
-            .intValue()
-        + ((Long) topicRequestsRepo.findAllRecordsCountForUserId(userId, tenantId).get(0)[0])
-            .intValue()
-        + ((Long) aclRequestsRepo.findAllRecordsCountForUserId(userId, tenantId).get(0)[0])
-            .intValue();
+    return calculateComponentsCountForUser(userId, tenantId, false);
```

Comment:
This now only seems to get called by tests, so if the true/false is not required it may be able to be simplified further always being lazy loaded.

ACLRequestsCountForCreatedStatus() Line 673 in AclRequestsIntegrationTests
and
deleteUserFailureisAdmin() line 294 in UsersTeamsControllerServiceTest.

So I think ACLRequestsCountForCreatedStatus() can be deleted and deleteUserFailureisAdmin() can be switched to use 
the new existsComponentsCountForUser.

then the existsOnly boolean can also be removed.

Reply:
Also sorry just completed the review, I should have added the other comment as a review.

Reply:
yeah...
however I noticed there is a more concise way to do it: replace `select all` with `select exists`
probably should be applied to this https://github.com/Aiven-Open/klaw/commit/1e9d6356c048cb50100c8194753f79a30dc7af42
as well